volatility3/framework/plugins/linux/vmayarascan.py

```diff
@@ -66,29 +71,36 @@ def _generator(self):
             # get the proc_layer object from the context
             proc_layer = self.context.layers[proc_layer_name]
 
-            for start, end in self.get_vma_maps(task):
-                for match in rules.match(
-                    data=proc_layer.read(start, end - start, True)
-                ):
-                    if yarascan.YaraScan.yara_returns_instances():
-                        for match_string in match.strings:
-                            for instance in match_string.instances:
-                                yield 0, (
-                                    format_hints.Hex(instance.offset + start),
-                                    task.UniqueProcessId,
-                                    match.rule,
-                                    match_string.identifier,
-                                    instance.matched_data,
-                                )
-                    else:
-                        for offset, name, value in match.strings:
-                            yield 0, (
-                                format_hints.Hex(offset + start),
-                                task.tgid,
-                                match.rule,
-                                name,
-                                value,
-                            )
+            vma_maps = list(self.get_vma_maps(task))
+            insane_vma_maps = [
+                start for (start, size) in vma_maps if size > sanity_check
+            ]
+            for start in insane_vma_maps:
+                vollog.debug(f"VMA at 0x{start:x} over sanity-check size, not scanning")
+
+            if not vma_maps:
+                vollog.warning(f"No VMAs were found for task {task.pid}, aborting")
+                continue
+
+            max_vma_size: int = max(
+                [size for (start, size) in vma_maps if size <= sanity_check]
+            )
+            scanner = yarascan.YaraScanner(rules=rules)
+            scanner.chunk_size = max_vma_size
+
+            # scan the process layer with the yarascanner
+            for offset, rule_name, name, value in proc_layer.scan(
+                context=self.context,
+                scanner=scanner,
+                sections=vma_maps,
```

Comment:
Your implementation includes a sanity check that filters based on `sanity_check` size, with a log message indicating those VMAs won't be scanned. However, it seems to be processing the entire `vma_maps` list regardless.. bug or I'm missing something here?

Reply:
Ah, good catch, I did indeed mean to pass in the filtered list.